modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/UserNotificationTypeTest.java

```diff
@@ -86,6 +91,80 @@ public void tearDown() throws PortalException {
 			user2.getUserId());
 	}
 
+	@Test
+	public void testIsDelivery() throws Exception {
+		NotificationTemplate notificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				_createNotificationContext(
+					Arrays.asList(
+						NotificationRecipientSettingUtil.
+							createNotificationRecipientSetting(
+								"userScreenName", user1.getScreenName())),
+					NotificationRecipientConstants.TYPE_USER));
+
+		ObjectAction objectAction = objectActionLocalService.addObjectAction(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			childObjectDefinition.getObjectDefinitionId(), true,
+			StringPool.BLANK, RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_NOTIFICATION,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_UPDATE,
+			UnicodePropertiesBuilder.put(
+				"notificationTemplateId",
+				notificationTemplate.getNotificationTemplateId()
+			).build(),
+			false);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user1);
+
+		List<NotificationQueueEntry> notificationQueueEntries =
+			notificationQueueEntryLocalService.getNotificationQueueEntries(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			notificationQueueEntries.toString(), 1,
+			notificationQueueEntries.size());
+
+		// Change UserNotificationDeliver
+
+		UserNotificationDelivery userNotificationDelivery =
+			_userNotificationDeliveryLocalService.fetchUserNotificationDelivery(
+				TestPropsValues.getUserId(),
+				childObjectDefinition.getPortletId(),
+				_classNameLocalService.getClassNameId(
+					childObjectDefinition.getClassName()),
+				UserNotificationDefinition.NOTIFICATION_TYPE_UPDATE_ENTRY,
+				UserNotificationDeliveryConstants.TYPE_WEBSITE);
+
+		_userNotificationDeliveryLocalService.updateUserNotificationDelivery(
+			userNotificationDelivery.getUserNotificationDeliveryId(), false);
+
+		_executeNotificationObjectAction(
+			notificationTemplate, childObjectDefinition, group.getGroupKey(),
+			user1);
+
+		notificationQueueEntries =
+			notificationQueueEntryLocalService.getNotificationQueueEntries(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+		Assert.assertEquals(
+			notificationQueueEntries.toString(), 1,
+			notificationQueueEntries.size());
```

Comment:
@mariuo I believe the assertion should be based on the `notificationRecipientSetting` ([this one](https://github.com/liferay-bpm/liferay-portal/pull/4547/files?diff=unified&w=0#diff-087a1582b1799639597a46d041570e3db330212e167869850889836f03df4dadR160)) because the notification queue entry is always added ([here](https://github.com/liferay-bpm/liferay-portal/pull/4547/files?diff=unified&w=0#diff-087a1582b1799639597a46d041570e3db330212e167869850889836f03df4dadR176-R177)) so we cannot ensure that the notification was sent or not based on the `notificationQueueEntry`, make sense? Maybe we should not create a `notificationQueueEntry` if an user is not notified and he is the only one recipient, but lets do the assertion based on the notificationRecipientSetting for now.

Thanks!